### PR TITLE
PHPUnit 10 Upgrade Warning

### PR DESCRIPTION
**Pull request by srtfisher**
CleanShot 2024-03-08 at 12 09 04@2x

Adds a warning during testing installation to alert the user about them attempting to run against a PHPUnit 10 code base.

### CODE REVIEW

File: src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+			return false;
+		}
+
+		return version_compare( '10.0.0', Version::id(), '<=' );
+	}
+
+	/**
+	 * Warn the user if they're running PHPUnit 10+ against an incompatible test suite.
+	 *
+	 * @param TestSuite $test_suite Test suite configuration.
```
Comment (attackant) [inline src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php:53]: Should this docblock have a return ?

File: src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	 */
+	protected function warn_if_test_suite_contains_legacy_test_cases( TestSuite $test_suite ): bool {
+		// Check if the test suite contains directories with a 'test-' prefix. That
+		// is a clear sign of a legacy test suite.
+		$legacy_test_cases = collect( $test_suite->directories() )
```
Comment (attackant) [inline src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php:58]: Will `test_suite` always have a `directories` method?

File: src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		}
+
+		if ( ! class_exists( Registry::class ) || ! method_exists( Registry::class, 'get' ) ) {
+			return;
+		}
+
+		$registry = Registry::get();
+
+		foreach ( $registry->testSuite()->asArray() as $test_suite ) {
+			if ( $test_suite instanceof TestSuite && $this->warn_if_test_suite_contains_legacy_test_cases( $test_suite ) ) {
```
Comment (attackant) [inline src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php:115]: `warn_if_test_suite_contains_legacy_test_cases` outputs a message and returns a boolean, is that a problem here? should those two concerns be separated into two functions?

File: src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php (hunk 1 above)
Comment (srtfisher) [reply src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php:53]: No. this repo doesn't use unnecessary returns

File: src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php (hunk 2 above)
Comment (srtfisher) [reply src/mantle/testing/concerns/trait-phpunit-upgrade-warning.php:63]: added a check

**Review by attackant — APPROVED** [review]
🐙

**Pull request merged**